datalayer-watch/src/main/java/com/google/android/horologist/datalayer/watch/WearDataLayerAppHelper.kt

```
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.horologist.datalayer.watch
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import androidx.wear.phone.interactions.PhoneTypeHelper
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.wearable.WearableStatusCodes
+import com.google.android.horologist.data.AppHelperResult
+import com.google.android.horologist.data.AppHelperResultCode
+import com.google.android.horologist.data.DataLayerAppHelper
+import com.google.android.horologist.data.ExperimentalHorologistDataLayerApi
+import com.google.android.horologist.data.companionConfig
+import com.google.android.horologist.data.launchRequest
+import kotlinx.coroutines.guava.await
+import kotlinx.coroutines.tasks.await
+
+/**
+ * Subclass of [DataLayerAppHelper] for use on Wear devices.
+ */
+@ExperimentalHorologistDataLayerApi
+public class WearDataLayerAppHelper(context: Context, private val appStoreUri: String? = null) :
+    DataLayerAppHelper(context) {
+    override suspend fun installOnNode(node: String) {
+        if (appStoreUri != null &&
```

Seems to silently do nothing in these cases, should we make that obvious via a return value or error?